test/jdk/java/awt/print/PrinterJob/PrintImage.java

```diff
@@ -21,276 +21,167 @@
-        private MenuItem        exitMenu = new MenuItem("Exit");
+    private final MenuItem print1Menu = new MenuItem("PrintTest1");
+    private final MenuItem print2Menu = new MenuItem("PrintTest2");
+    private final MenuItem exitMenu = new MenuItem("Exit");
 
-        public static void main(String[] argv) {
-        String[] instructions =
-           { "You must have a printer available to perform this test,",
-             "prefererably Canon LaserShot A309GII.",
-             "Printing must be done in Win 98 Japanese 2nd Edition.",
-             "",
-             "Passing test : Output of text image for PrintTest1 and PrintTest2 should be same as that on the screen.",
-           };
+    private static final String INSTRUCTIONS =
+            "You must have a printer available to perform this test,\n" +
+            "prefererably Canon LaserShot A309GII.\n" +
+            "Printing must be done in Win 98 Japanese 2nd Edition.\n" +
+            "\n" +
+            "Passing test : Output of text image for PrintTest1 and PrintTest2 should be " +
+            "same as that on the screen.";
 
-        Sysout.createDialog( );
-         Sysout.printInstructions( instructions );
+    public static void main(String[] argv) throws Exception {
 
-                new PrintImage();
+        if (PrinterJob.lookupPrintServices().length == 0) {
+            throw new SkippedException("Printer not configured or available."
+                    + " Test cannot continue.");
         }
 
-        public PrintImage() {
-                super("PrintImage");
-                initPrintImage();
-        }
+        PassFailJFrame.builder()
+                .instructions(INSTRUCTIONS)
+                .testUI(PrintImage::new)
+                .rows((int) INSTRUCTIONS.lines().count() + 1)
+                .columns(45)
+                .build()
+                .awaitAndCheck();
+    }
 
-        public void initPrintImage() {
+    public PrintImage() {
+        super("PrintImage");
+        initPrintImage();
+    }
 
-                printImageCanvas = new PrintImageCanvas(this);
+    public void initPrintImage() {
 
-                initMenu();
+        printImageCanvas = new PrintImageCanvas(this);
 
-                addWindowListener(new WindowAdapter() {
-                        public void windowClosing(WindowEvent ev) {
-                                dispose();
-                        }
-                        public void windowClosed(WindowEvent ev) {
-                                System.exit(0);
-                        }
-                });
+        initMenu();
 
-                setLayout(new BorderLayout());
-                add(printImageCanvas, BorderLayout.CENTER);
-                pack();
+        setLayout(new BorderLayout());
+        add(printImageCanvas, BorderLayout.CENTER);
+        pack();
 
-                setSize(500,500);
-                setVisible(true);
-        }
+        setSize(500, 500);
```

Comment:
You either `pack` to use the preferred sizes of all the components or `setSize`.